pytorch_lightning/trainer/progress.py

```diff
@@ -141,22 +141,22 @@ def scheduler_steps(self) -> int:
 
 
 @dataclass
-class TrainingProgress(Progress):
-    """
-    Extends ``Progress`` with training specific attributes
-
-    Args:
-        optimization: Tracks optimization progress
-    """
-    optimization: OptimizationProgress = field(default_factory=OptimizationProgress)
+class TrainingLoopProgress(LoopProgress):
 
+    optimizer_idx: Optional[int] = field(default_factory=int)
+    epoch: Progress = field(default_factory=Progress)
+    optimizations: Optional[tuple] = field(default_factory=tuple)
```

Comment:
It's probably okay to keep `optimizer_idx` here since it's a property tied to the `TrainingLoop`, however, `optimizations` should be part of `TrainingProgress`

Reply:
Yes, I agree.